lisp/ess-r-xref.el

```diff
@@ -65,6 +74,27 @@
       (goto-char (match-beginning 0))
       (read (current-buffer)))))
 
+(defun ess-r-xref--pkg-srcfile (symbol r-src-file)
+  "Look in the source directory of the R package containing symbol SYMBOL for R-SRC-FILE."
+  (let* ((env-name (ess-string-command (format ".ess_pkg_name(\"%s\")\n" symbol)))
+         (pkg (if (string-equal env-name "R_GlobalEnv")
+                  ;; This probably shouldn't happen, but just in case...
+                  (error "Can't find package for symbol %s." symbol)
+                env-name))
+         (dir (or (assoc-default pkg ess-r-xref-pkg-sources)
+                  (when ess-r-xref-pkg-source-dir
+                    (expand-file-name pkg ess-r-xref-pkg-source-dir))
+                  (read-directory-name
+                   (format "%s package source dir: " pkg) nil nil t)))
+         (file (expand-file-name r-src-file dir)))
+    (if (file-readable-p file)
```

Comment:
To clarify: you want me to locate the file in the R session, as opposed to Emacs, so it can handle remote files? While I think it would be nice to handle remote files, the reason I don't think this make senses for now is that we'd also need to make `xref-file-location` handle non-local files in the process. Are there mechanisms in place in ESS to do that already?

Reply:
No mechanisms in ESS, those are handled transparently by tramp.  I presume xref should work out of the box with tramp, but who knows. These are troubles for later when we start dealing with remotes. For now, I mainly want to be consistent; the other check in ess-r-xref.el uses R session for checking file existence.

Reply:
Ok, then maybe the right thing to do is actually change the other code. I agree that using tramp is where we want to end up.